File: spring-web-java/src/main/java/spring/web/java/global/converter/persistence/LocalDateTimeConverter.java
package spring.web.java.global.converter.persistence;

import java.time.LocalDateTime;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import spring.web.java.global.common.DateTimeUtils;

@Converter(autoApply = true)
public class LocalDateTimeConverter implements AttributeConverter<LocalDateTime, LocalDateTime> {

	@Override
	public LocalDateTime convertToDatabaseColumn(LocalDateTime attribute) {
		if (attribute == null) {
			return null;
		}
		return DateTimeUtils.convertKstToUtc(attribute);
	}

	@Override
	public LocalDateTime convertToEntityAttribute(LocalDateTime dbData) {
		if (dbData == null) {
			return null;
		}
		return DateTimeUtils.convertUtcToKst(dbData);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/converter/persistence/LocalDateTimeConverter.kt
package spring.web.kotlin.global.converter.persistence

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import spring.web.kotlin.global.common.convertToKst
import spring.web.kotlin.global.common.convertToUtc
import java.time.LocalDateTime

@Converter(autoApply = true)
class LocalDateTimeConverter: AttributeConverter<LocalDateTime?, LocalDateTime?> {

    override fun convertToDatabaseColumn(attribute: LocalDateTime?) = attribute?.convertToUtc()

    override fun convertToEntityAttribute(dbData: LocalDateTime?) = dbData?.convertToKst()
}